model/roles.go

```diff
@@ -168,26 +311,80 @@ func (r *Role) GetScriptPaths() map[string]string {
 
 }
 
-// GetRoleDevVersion gets the aggregate signature of all jobs and packages
-func (r *Role) GetRoleDevVersion() string {
-	roleSignature := ""
+// func writeScripts(hasher hash.Hash, parentDir string, scripts []string, label string) {
+func writeScripts(hasher io.Writer, parentDir string, scripts []string, label string) {
+	io.WriteString(hasher, "<"+label+">")
+	for _, script := range scripts {
+		io.WriteString(hasher, script)
+		io.WriteString(hasher, ":\n")
+		contents, err := ioutil.ReadFile(filepath.Join(parentDir, script))
+		if err != nil {
+			// Don't bother emitting an error message as the verifier should have checked
+			// the existence of all scripts.
+			// Also post_config_scripts are given in terms of destination container paths,
+			// not source paths, so we don't have them anyway.
+			continue
+		}
+		hasher.Write(contents)
+	}
+	io.WriteString(hasher, "</"+label+">")
+}
+
+// GetRoleDevVersion gets the aggregate signature of all jobs, packages, scripts, and properties
+func (r *Role) GetRoleDevVersion() (string, error) {
 	var packages Packages
 
+	globalConfig := r.rolesManifest.globalConfig
+	if globalConfig == nil {
+		return "", fmt.Errorf("Need to call roleManifest.SetGlobalConfig")
+	}
+	globalDefaultValues := globalConfig.globalDefaultValues
+	lightOpinions := globalConfig.lightOpinions
+	darkOpinions := globalConfig.darkOpinions
+	localDefaultValues := getDefaultValuesHash(r.Configuration.Templates)
+	scriptParentDir := filepath.Dir(r.rolesManifest.manifestFilePath)
+	hasher := sha1.New()
+	io.WriteString(hasher, r.Name)
+	io.WriteString(hasher, r.Type)
+	writeScripts(hasher, scriptParentDir, r.EnvironScripts, "EnvironScripts")
+	writeScripts(hasher, scriptParentDir, r.Scripts, "Scripts")
+	writeScripts(hasher, scriptParentDir, r.PostConfigScripts, "PostConfigScripts")
+	io.WriteString(hasher, "<jobs>")
 	// Jobs are *not* sorted because they are an array and the order may be
 	// significant, in particular for bosh-task roles.
 	for _, job := range r.Jobs {
-		roleSignature = fmt.Sprintf("%s\n%s", roleSignature, job.SHA1)
+		io.WriteString(hasher, job.SHA1)
 		packages = append(packages, job.Packages...)
+		hasher.Write([]byte("<properties>"))
+		for _, property := range job.Properties {
```

Comment:
Within a single job, does the order of properties matter ?
If not we should sort them here, I think.